vllm/core/scheduler.py

```diff
@@ -1202,9 +1202,9 @@ def _can_append_slots(self, seq_group: SequenceGroup,
             seq_group=seq_group, num_lookahead_slots=num_lookahead_slots)
 
     def _allow_async_output_proc(self, seq_group: SequenceGroup) -> bool:
+        # TODO: does it work with parallel sampling?
```

Comment:
@megha95 does async output processing work with parallel sampling?

if yes, after removing beam search, we can also remove this function.

Reply:
@youkaichao no it does not. Scheduler switches back to sync mode when it sees any request with `best_of != 1`